core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1050,7 +1050,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
    * }}}
    */
   final def resurrect(implicit ev1: E <:< Throwable): RIO[R, A] =
-    self.sandbox.mapError(_.squash)
+    self.foldCauseM(cause => ZIO.fail(cause.squash), ZIO.succeedNow)
```

Comment:
`unrefineWith { case e => e }(ev1)`

Reply:
Not sure, `unrefineWith` due to the partial is doing a `find` and a recursive `fold`, instead of a direct call to `foldCauseM`.

Reply:
That's fine.